sky/packages/sky/lib/src/fn3/scrollable.dart

```diff
@@ -207,8 +207,56 @@ abstract class ScrollableState<T extends Scrollable> extends State<T> {
   }
 }
 
-// TODO(abarth): findScrollableAncestor
-// TODO(abarth): ensureWidgetIsVisible
+ScrollableState findScrollableAncestor(BuildContext context) {
+  ScrollableState result;
+  context.visitAncestorElements((Element element) {
+    if (element is StatefulComponentElement) {
+      if (element.state is ScrollableState) {
+        result = element.state;
+        return false;
+      }
+    }
+    return true;
+  });
+  return result;
+}
+
+Future ensureWidgetIsVisible(BuildContext context, { Duration duration, Curve curve }) {
+  assert(context.findRenderObject() is RenderBox);
+
+  ScrollableState scrollable = findScrollableAncestor(context);
+  if (scrollable == null)
+    return new Future.value();
+
+  RenderBox targetRenderObject = context.findRenderObject();
+  assert(targetRenderObject.attached);
+  Size targetSize = targetRenderObject.size;
+  Point targetCenter = targetRenderObject.localToGlobal(
+    scrollable.config.scrollDirection == ScrollDirection.vertical
```

Comment:
use a switch when dealing with an enum, since that way the compiler warns you that you missed a case.

Like here, where you missed "both". (Just assert and point to #888 for now.)